CMakeLists.txt

```diff
@@ -0,0 +1,162 @@
+cmake_minimum_required(VERSION 3.12)
+
+project(wannier90 VERSION 4.0.0 LANGUAGES Fortran) # welcome to Wannier90!
+
+include(GNUInstallDirs)
+
+# -Dwith_mpi=off for serial build (default MPI)
+# otherwise:
+# -Dwith_mpif08=off to disable use of f08 MPI module, to use f90 module instead
+# -Dwith_mpih=on to force use of mpif.h include file
+
+option(with_mpi "build MPI version" ON)
+if (with_mpi)
+  option(with_mpif08 "build MPI version using f08 module" ON)
+  option(with_mpih "build MPI version using Fortran 77 mpif.h header" OFF)
```

Comment:
Options should not be if blocked against other option, they are not properly parsed. Also why not fully remove `mpih` support in 2023? Legacy systems should use legacy code or upgrade